### ISSUE TITLE
Remove well-parenthesized structure other than just definitions

### ISSUE THREAD

**Issue by JasonGross** [JasonGross/coq-tools#89]
Maybe after #56, it would be nice to try removing well-balanced structures (sections, modules, module types, etc) before we try to remove individual lines in them.

**Comment by ejgallego** [comment]
Indeed, actually related to the discussion at https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs.20.26.20plugin.20devs/topic/Abort.20for.20modules.2Fprogram.3F , I do have something that could help.

Even if you get access to the Coq's AST, it is still mostly a sequence of commands and the clients have to do significant guesswork to understand the structure.

However, I have implemented a structured presentation of documents which does exactly this, so we could expose it for your comsuption.
